src/overlays/actors/ovl_Bg_Spot02_Objects/z_bg_spot02_objects.c

```diff
@@ -28,29 +48,287 @@ const ActorInit Bg_Spot02_Objects_InitVars = {
+        }
+    }
+}
+
+void func_808ACA08(BgSpot02Objects* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (this->timer == 20) {
+        this->dyna.actor.draw = NULL;
+        EffectSsHahen_SpawnBurst(globalCtx, &this->dyna.actor.world.pos, 30.0f, 0, 25, 5, 40, OBJECT_SPOT02_OBJECTS, 20,
+                                 D_06012D30);
+    } else if (this->timer == 0) {
+        Actor_Kill(&this->dyna.actor);
+    }
+
+    if (globalCtx->csCtx.frames == 402) {
+        if (LINK_IS_CHILD) {
+            func_8002F7DC(&player->actor, NA_SE_VO_LI_DEMO_DAMAGE_KID);
+        } else {
+            func_8002F7DC(&player->actor, NA_SE_VO_LI_DEMO_DAMAGE);
+        }
+    }
+}
+
+void func_808ACAFC(BgSpot02Objects* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->unk_16B)) {
+        Actor_SetFocus(&this->dyna.actor, 60.0f);
+        func_80080480(globalCtx, &this->dyna.actor);
+        this->actionFunc = func_808ACB58;
+    }
+}
+
+void func_808ACB58(BgSpot02Objects* this, GlobalContext* globalCtx) {
+    if (Math_StepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y + 255.0f, 1.0f)) {
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_STONEDOOR_STOP);
+        this->actionFunc = func_808AC8FC;
+    } else {
+        func_8002F974(&this->dyna.actor, NA_SE_EV_WALL_MOVE_SP - SFX_FLAG);
+    }
+}
+
+void BgSpot02Objects_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot02Objects* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void BgSpot02Objects_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Gfx* dLists[] = { 0x06012A50, 0x060127C0, 0x060130B0 };
+
+    Gfx_DrawDListOpa(globalCtx, dLists[thisx->params]);
+}
+
+void func_808ACC34(BgSpot02Objects* this, GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state != 0 && globalCtx->csCtx.npcActions[0] != NULL &&
+        globalCtx->csCtx.npcActions[0]->action == 2) {
+        this->unk_16A++;
+
+        if (this->unk_16A >= 12) {
+            Actor_Kill(&this->dyna.actor);
+        }
+    }
+
+    if (globalCtx->csCtx.frames == 245 || globalCtx->csCtx.frames == 351) {
+        func_800788CC(NA_SE_EV_LIGHTNING);
+    }
+}
+
+void func_808ACCB8(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot02Objects* this = THIS;
+    f32 pad;
+    f32 pad2;
+    u8 redPrim;
+    u8 greenPrim;
+    u8 bluePrim;
+    u8 redEnv;
+    u8 greenEnv;
+    u8 blueEnv;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_spot02_objects.c", 600);
+
+    if (globalCtx->csCtx.state != 0 && globalCtx->csCtx.npcActions[0] != NULL &&
+        globalCtx->csCtx.npcActions[0]->action == 2) {
+        if (this->unk_16A < 5) {
+            pad = (this->unk_16A / 5.0f);
+            redPrim = greenPrim = bluePrim = 255;
+            redEnv = 100.0f + 155.0f * pad;
+            greenEnv = 255;
+            blueEnv = 255.0f - 155.0f * pad;
```

Comment:
I wonder if this should be expressed like this instead: 
```suggestion
            redEnv = 100 + ((255-100) * pad);
            greenEnv = 255;
            blueEnv = 255 + ((100-255) * pad);
```